catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/LookController.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui.recyclerview.controller
+
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.common.LookData
+import org.catrobat.catroid.content.Scene
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.io.StorageOperations
+import org.catrobat.catroid.ui.controller.BackpackListManager
+import org.catrobat.catroid.ui.recyclerview.util.UniqueNameProvider
+import org.catrobat.catroid.utils.Utils
+import java.io.File
+import java.io.IOException
+
+class LookController {
+    private val uniqueNameProvider = UniqueNameProvider()
+    @Throws(IOException::class)
+    fun copy(lookToCopy: LookData?, dstScene: Scene?, dstSprite: Sprite?): LookData {
+        val name = uniqueNameProvider.getUniqueNameInNameables(lookToCopy?.name, dstSprite?.lookList)
+        val dstDir = dstScene?.let { getImageDir(it) }
+        val file = StorageOperations.copyFileToDir(lookToCopy?.file, dstDir)
+        return LookData(name, file)
+    }
+
+    @Throws(IOException::class)
+    fun findOrCopy(lookToCopy: LookData?, dstScene: Scene?, dstSprite: Sprite?): LookData {
+        dstSprite?.lookList?.forEach {
+            if (compareByChecksum(it.file, lookToCopy?.file)) {
+                return it
+            }
+        }
```

Comment:
Please use https://kotlinlang.org/docs/collection-elements.html#retrieve-by-condition.
Since it is used three times in this file, extract it, please.

Reply:
You can also make compareByChecksum to a private extension Function if you like ;)

Reply:
```suggestion
        val exisingCopy = dstSprite?.lookList?.firstOrNull(it -> it.file.checksumEquals(lookToCopy?.file))
        if (exisingCopy != null) return exisingCopy
        ....
        private File.compareByChecksum(file :File?) { .... }
```
https://kotlinlang.org/docs/extensions.html

Reply:
Please rename compareChecksum to checksumEquals or smth similar